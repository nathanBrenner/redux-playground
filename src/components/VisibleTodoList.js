import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../actions";
import TodoList from "./TodoList";
import { withRouter } from "react-router";
import { getVisibleTodos } from "../reducers";

class VisibleTodoList extends Component {
  componentDidMount() {
    this.fetchData();
  }

  componentDidUpdate(prevProps) {
    if (this.props.filter !== prevProps.filter) {
      this.fetchData();
    }
  }
  render() {
    const { toggleTodo, ...rest } = this.props;
    return <TodoList {...rest} onTodoClick={toggleTodo} />;
  }

  fetchData() {
    const { filter, fetchTodos } = this.props;
    fetchTodos(filter);
  }
}

const mapStateToProps = (state, { match: { params } }) => {
  const filter = params.filter || "all";
  return {
    todos: getVisibleTodos(state, filter),
    filter
  };
};

VisibleTodoList = withRouter(
  connect(mapStateToProps, actions)(VisibleTodoList)
);

export default VisibleTodoList;
