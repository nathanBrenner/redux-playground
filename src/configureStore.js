import todoApp from "./reducers";
import { createStore, applyMiddleware } from "redux";
import promise from "redux-promise";
import { createLogger } from "redux-logger";

const configureStore = () => {
  const middlewares = [promise];

  if (process.env.NODE_ENV !== "production") {
    middlewares.push(createLogger());
  }

  const store = createStore(
    todoApp,
    window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(...middlewares)
  );
  //     window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  return store;
};

export default configureStore;
