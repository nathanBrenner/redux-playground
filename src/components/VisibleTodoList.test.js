import React from "react";
import VisibleTodoList from "./VisibleTodoList";
import configureStore from "../configureStore";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";

describe("VisibleTodoList", () => {
  test("renders without crashing", () => {
    const filter = "all";
    const todos = [{ id: "1", completed: false, text: "do something" }];
    const isFetching = false;
    const fetchTodo = () => true;
    const toggleTodo = () => true;
    const store = configureStore();
    const props = { filter, todos, isFetching, fetchTodo, toggleTodo, store };
    shallow(<VisibleTodoList {...props} />);
  });

  test("snapshot", () => {
    const filter = "all";
    const todos = [{ id: "1", completed: false, text: "do something" }];
    const isFetching = false;
    const fetchTodo = () => true;
    const toggleTodo = () => true;
    const store = configureStore();
    const props = { filter, todos, isFetching, fetchTodo, toggleTodo, store };
    const component = renderer.create(
      <MemoryRouter>
        <VisibleTodoList {...props} />
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
