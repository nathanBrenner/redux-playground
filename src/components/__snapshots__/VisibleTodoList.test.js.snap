// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisibleTodoList snapshot 1`] = `
<p>
  Loading...
</p>
`;
